### PR TITLE
CB-60: add posibility to remove recipe, add global alert

### PR DESCRIPTION
**Pull request by bartekguz**
https://team-1626248039387.atlassian.net/jira/software/projects/CB/boards/1?selectedIssue=CB-60

Add possibility to remove recipe.
Add global alert `react-tostify`

### CODE REVIEW

File: src/pages/RecipeDetails/RecipeDetails.tsx [hunk 1]
```diff
@@ -14,34 +14,71 @@ import {
...
+  const deleteData = async () => {
+    try {
+      await deleteRecipe(id);
+      handleCloseDialog();
+      toast.success('Recipe deleted successfully!', {
+        position: toast.POSITION.BOTTOM_CENTER,
+      });
+      history.push('/');
+    } catch {
+      console.warn('Cannot delete recipe.');
```
Comment (Aliath) [inline src/pages/RecipeDetails/RecipeDetails.tsx:63]: `toast.error`?

**Review by Aliath — CHANGES_REQUESTED** [review]

File: src/hooks/useAuth/useAuth.ts [hunk 2]
```diff
@@ -23,6 +24,10 @@ export const useAuth = () => {
       history.push('/');
     } catch {
       setRequestState('error');
+      toast.error('Login failed!', {
```
Comment (Aliath) [inline src/hooks/useAuth/useAuth.ts:27]: Do not shout at user 😉

File: src/pages/RecipeDetails/RecipeDetails.tsx [hunk 3]
```diff
@@ -14,34 +14,71 @@ import {
...
+  DialogContent,
+  DialogActions,
 } from '@material-ui/core';
 import './RecipeDetails.scss';
+import { RequestState } from 'types/request';
+import { TransitionProps } from '@material-ui/core/transitions';
+import { toast } from 'react-toastify';
 
-type RequestState = 'loading' | 'loaded' | 'failure';
+const Transition = React.forwardRef(function Transition(
```
Comment (Aliath) [inline src/pages/RecipeDetails/RecipeDetails.tsx:29]: `const Transition = ...(function Transition(`

Something is wrong right here 😄

File: src/pages/RecipeDetails/RecipeDetails.tsx [hunk 4]
```diff
@@ -14,34 +14,71 @@ import {
...
+) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
 
 export const RecipeDetails: React.FC = () => {
   const [recipe, setRecipe] = useState<RecipeWithDetails | undefined>(
     undefined
   );
   const [currentState, setCurrentState] = useState<RequestState>('loading');
+  const [open, setOpen] = useState(false);
```
Comment (Aliath) [inline src/pages/RecipeDetails/RecipeDetails.tsx:41]: What is open?

File: src/pages/RecipeDetails/RecipeDetails.tsx [hunk 5]
```diff
@@ -14,34 +14,71 @@ import {
...
+  };
+
+  const deleteData = async () => {
+    try {
+      await deleteRecipe(id);
+      handleCloseDialog();
+      toast.success('Recipe deleted successfully!', {
+        position: toast.POSITION.BOTTOM_CENTER,
+      });
+      history.push('/');
```
Comment (Aliath) [inline src/pages/RecipeDetails/RecipeDetails.tsx:61]: What about `history.goBack` or `/recipes`?

File: src/pages/RecipeDetails/RecipeDetails.tsx [hunk 6]
```diff
@@ -120,6 +159,31 @@ export const RecipeDetails: React.FC = () => {
           <RecipeComments comments={recipe.comments} />
         </Card>
       )}
+      <Dialog
+        open={open}
+        TransitionComponent={Transition}
+        keepMounted
+        onClose={handleCloseDialog}
+        aria-labelledby="alert-dialog-slide-title"
```
Comment (Aliath) [inline src/pages/RecipeDetails/RecipeDetails.tsx:167]: `recipe-delete-dialog-title`

File: src/pages/RecipeDetails/RecipeDetails.tsx [hunk 7]
```diff
@@ -120,6 +159,31 @@ export const RecipeDetails: React.FC = () => {
...
+        aria-describedby="alert-dialog-slide-description"
```
Comment (Aliath) [inline src/pages/RecipeDetails/RecipeDetails.tsx:168]: `recipe-delete-dialog-description`

File: src/pages/RecipeDetails/RecipeDetails.tsx (hunk 3 above)
Comment (bartekguz) [reply src/pages/RecipeDetails/RecipeDetails.tsx:29]: Do you mean get rid of `React.`?

Comment (Aliath) [reply src/pages/RecipeDetails/RecipeDetails.tsx]: That's another thing but you define function `Transition` twice - the first time by `function Transition` and the second by assignment `forwardRef` result.

Comment (bartekguz) [reply src/pages/RecipeDetails/RecipeDetails.tsx]: Hmm... as before this is the code from the material-ui example: 
https://material-ui.com/components/dialogs/#AlertDialogSlide.tsx

I tried to define this function as you wrote - one time and it doesn't work.
Can you give me an example how to do it?

Comment (Aliath) [reply src/pages/RecipeDetails/RecipeDetails.tsx]: Ok, it's all about the content of `Transition` component. In general it is not a good practice to define different things under the same name. In this case we want to pass reference via `forwardRef` and the problem is eslint that requires name for wrapped function.

to summary: there are three solutions
a) ignore it and just keep the `Transition` as it is right now
b) rename `const Transition` to `const TransitionWithRef`
c) use eslint ignore to calm down the linter

pick whatever you want :D

**Review by Aliath — APPROVED** [review]
🎸 👋

**Pull request merged**